.github/workflows/dispatch_analytics.yml

```diff
@@ -28,6 +26,20 @@ jobs:
       group: analytics
     runs-on: ubuntu-latest
     environment: analytics
+    permissions:
```

Comment:
Is the level of permission block affecting the score card?

Reply:
Yep, check [this section](https://github.com/ossf/scorecard/blob/main/docs/checks.md#token-permissions) from their docs, the relevant sentence is this:

> The highest score is awarded when the permissions definitions in each workflow's yaml file are set as read-only at the [top level](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#permissions) and the required write permissions are declared at the [run-level](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idpermissions). One point is reduced from the score if all jobs have their permissions defined but the top level permissions are not defined. This configuration is secure, but there is a chance that when a new job is added to the workflow, its job permissions could be left undefined because of human error.

So basically we need to set a permission at the file (aka workflow) level that is limited (`read`) and then define appropriate permissions at the job level. This way other jobs can't inadvertently access data they're not supposed to.

Reply:
I should also note that I'm still getting up to speed with the scorecard system, for now I'm using the Py repo as baseline since I confirmed with them that they don't have findings in these areas.

Reply:
that makes sense, I just wasn't sure if it applies for only one job in a workflow, but I see the value in scoping it down to individual job.